components/sock_utils/idf_component.yml

```diff
@@ -1,4 +1,4 @@
-version: 0.1.0
+version: 0.2.0
 description: The component provides helper implementation of common system/socket utilities
 url: https://github.com/espressif/esp-protocols/tree/master/components/sock_utils
 dependencies:
```

Comment:
Commenting on the lines below: looks like the dependency should now be on IDF >= 5.5, since IDF 5.0 doesn't know about LWIP_SOCKET_HAS_SOCKETPAIR.

Reply:
Added additional notes to the component README:

**Notes**:

- **`*)`** `socketpair()` and `pipe()` are built on top of `lwIP` TCP sockets, inheriting the same characteristics. For instance, the maximum transmit buffer size is based on the `TCP_SND_BUF` setting.
- **`**)`** `socketpair()` and `gai_strerror()` are declared in sock_utils header files, the declaration is propagated to ESP-IDF from v5.5 to the official header files. If you're using older IDF version, you need to manually pre-include related header files from the sock_utils public include directory.
- **`***)`** `pipe()` is declared in compiler's `sys/unistd.h`.

PTAL again @igrr if it addresses your concern

Reply:
> I mean there is a certain value of using this component even if LWIP_SOCKET_HAS_SOCKETPAIR isn't propagated.

Okay, I agree with that. I think with the notes added in the README this does address my comment, thank you!

Reply:
Maybe just one nitpick,

> declared in sock_utils header files, the declaration is propagated to ESP-IDF from v5.5 to the official header files

could you mention the names of the header files for easier searching?

Reply:
they're listed in table itself, last column: `declared in` (this is only the note below the text)

https://github.com/espressif/esp-protocols/blob/5dc38a88ad73807a7ae9b0c586116f1c8ebf10fc/components/sock_utils/README.md?plain=1#L12

Reply:
Right, I'm sorry, I have overlooked that part. Everything is great now.